Packages/Primitives/Sources/Extensions/BigInt+Primitives.swift

```diff
@@ -28,6 +28,11 @@ public extension BigInt {
         return BigInt(result.rounded())
     }
     
+    func multiply(byPercentage percentage: Double) -> BigInt {
+        let result = Double(self) * percentage
+        return BigInt(result.rounded())
```

Comment:
```suggestion
        BigInt(Double(self) * percentage).rounded()
```

if you had to round. much cleaner